### PR TITLE
[TTAHUB-1759] Training report form

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change
This PR includes
- all instances of react-hook-form have been bumped to use the non IE-11 distro
- a new permission was added "COLLABORATOR_TRAINING_REPORTS"
- Navigator component refactored to activity-report-agnostic
- First page of form created ("event summary")

This does not include:
- exhaustive permissions. we can rely on the feature flag for now, as this PR is already large enough
- good test coverage. will wait till more functionality is complete to add additional tests, like e2e 

I also understand that this may need to be revised once #1558 is merged. Planning on crossing that bridge when I come to it.

## How to test
- This PR includes updates to the ```@ttahub/common``` package, so you'll need to reinstall that
- Import training events via the importCLI job (you may have to modify the script with an existing email if you aren't using the test data)
- You may also need to add users with the new permissions
- View the form at /training-report/{:trainingReportId}
- Note the imported fields, make changes, save draft. Confirm that the handlers work.
- Smoke test the Activity report form
- **The main thing to test on this PR is that the ActivityReport form remains intact and functional without any new bugs.**

Note also - as I've stated, due to the size of this PR, you should expect things to be incomplete. Review #1577 for a more functional version of this. This is intended to be a first step, the separation of the Navigator component from the AR without breaking anything.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1759
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1769

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

**Review by AdamAdHocTeam — COMMENTED** [review]
Overall looks good was able to test the old report page. Also update an existing event in the db.

Two things:
1. I couldn't get start date to persists on the new form.
2. I noticed we track the for reasons separate from the 'Reason for Activity'. I assume these are different?

**Review by AdamAdHocTeam — APPROVED** [review]
My issues have been fixed. 

Looks like a good starting point.

File: src/routes/users/index.js [hunk 1]
```diff
@@ -19,7 +20,7 @@ router.get('/collaborators', transactionWrapper(getPossibleCollaborators));
 router.get('/stateCodes', transactionWrapper(getPossibleStateCodes));
 router.get('/statistics', transactionWrapper(getUserStatistics));
 router.get('/active-users', transactionWrapper(getActiveUsers));
-
+router.get('/training-report-users', transactionWrapper(getTrainingReportUsers)); // todo - update redoc with new route
```
Comment (GarrettEHill) [inline src/routes/users/index.js:23]: add e2e-api test for new route

File: src/tools/importSmartSheetEvent.test.js [hunk 2]
```diff
@@ -82,33 +80,33 @@ describe('Import Smart Sheet Events', () => {
...
+        eventName: 'Health Webinar Series: Oral Health and Dental Care from a Regional and State Perspective',
+        eventOrganizer: 'Regional PD Event (with National Centers)',
+        audience: 'Recipients',
         'Event Duration/# NC Days of Support': 'Series',
-        'Reason for Activity': 'Ongoing Quality Improvement',
-        'Target Population(s)': 'None',
-        'Overall Vision/Goal for the PD Event': 'Oral Health',
-        Creator: 'smartsheetevents@ss.com',
+        reasons: ['Ongoing Quality Improvement'],
+        targetPopulations: ['None'],
```
Comment (GarrettEHill) [inline src/tools/importSmartSheetEvent.test.js:107]: One of the test should exercise the splitArrayTransformer function on either reasons, targetPopulations or both.

File: src/tools/importSmartSheetEvent.test.js [hunk 3]
```diff
@@ -82,33 +80,33 @@ describe('Import Smart Sheet Events', () => {
...
+        vision: 'Oral Health',
+        creator: 'smartsheetevents@ss.com',
```
Comment (GarrettEHill) [inline src/tools/importSmartSheetEvent.test.js:109]: what is the plan when/if this email does not match one in our users table?

**Review by GarrettEHill — COMMENTED** [review]
just some small things

File: src/tools/importSmartSheetEvent.test.js (hunk 3 above)
Comment (thewatermethod) [reply src/tools/importSmartSheetEvent.test.js:109]: It doesn't import the event.

While the script did not originate with this PR, I believe Adam vetted this in the original implementation.

File: src/routes/users/index.js (hunk 1 above)
Comment (thewatermethod) [reply src/routes/users/index.js:23]: https://github.com/HHS/Head-Start-TTADP/pull/1559/commits/1c4e1a6469d7372c4bc9e07bb013684a998a98af

File: src/tools/importSmartSheetEvent.test.js (hunk 2 above)
Comment (thewatermethod) [reply src/tools/importSmartSheetEvent.test.js:109]: https://github.com/HHS/Head-Start-TTADP/pull/1559/commits/0c0fe1e9a44f69f0b3fbf7e8d5b907110920674c

**Review by kryswisnaskas — COMMENTED** [review]
Big PR! 🚀 
The activity report seemed to be working fine. 👍  I was also able to use the training form 👍 

Other comments included.

File: frontend/src/pages/TrainingReportForm/index.js [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+
+  return (
+    <div className="smart-hub-training-report">
+      { error
+      && (
+      <Alert type="warning">
+        {error}
+      </Alert>
+      )}
+      <Helmet titleTemplate="%s - Activity Report - TTA Hub" defaultTitle="TTA Hub - Activity Report" />
```
Comment (nvms) [inline frontend/src/pages/TrainingReportForm/index.js:266]:
```suggestion
      <Helmet titleTemplate="%s - Training Report - TTA Hub" defaultTitle="TTA Hub - Training Report" />
```

**Review by nvms — APPROVED** [review]
really good start of the TR form. nice refactoring of things and I like the addition of NavigatorButtons.

**Pull request merged**